content/ADR-189-social-service-M2.md

```diff
@@ -0,0 +1,166 @@
+---
+layout: adr
+adr: 189
+title: Social Service M2
+date: 2023-02-08
+status: Draft # pick one of these
+type: RFC # pick one of these
+spdx-license: CC0-1.0
+authors:
+  - jmoguilevsky
+  - agusaldasoro
+---
+
+## Abstract
+
+A document that compares two possible solutions to stop leveraging Matrix to handle friend requests amongst users. To achieve this, the service will need to not only save the friendship interactions but also have a mechanism to notify online users in real-time of the interactions.
+The chosen mechanism will be WebSockets.
+
+## Context, Reach & Prioritization
+
+Currently, all friendship events are being stored in both Matrix and the Social Service, to avoid this duplication of information this RFC proposes to start handling friendship requests directly on the Social Service and stop storing the information in Matrix. The benefits of this proposal include:
+
+- Downscale the load on Matrix
+  - The friendships events (Request, Accept, Delete, etc.)
+  - Prevent creating rooms when a friendship hasn't been established
+- Be able to apply friend request policies (i.e. prevent spamming, max amount of friends, blocking, etc.)
+- Extend the friendship API to know the pending friend requests a user has
+
+Also, this new approach will need to be able to notify the users in real-time about the new friendship events. A new communication system will be needed to established to connect the client and the server (the technical specification of this system will be explored later).
+
+## Solution Space Exploration
+
+There are two main approaches:
+
+### WebSocket / Long polling
+
+WebSocket inherently comes with the benefit of establishing a bidirectional connection, enabling more features for the future (i.e. presence). Another big plus of using WebSocket is that the client can leverage protobuf and RPC for communication between the server and the client.
+On the other hand, WebSocket + RPC implies that the Social server will need to be the first implementation of Rust-RPC which is not yet a production-tested product.
```

Comment:
Why do you use"Rust-RPC" instead of "rpc-rust" (repo name) or "dcl-rpc" (crate name), or also a link to the project?